src/main/java/ch/njol/skript/effects/EffLoadServerIcon.java

```diff
@@ -0,0 +1,98 @@
+/**
+ *   This file is part of Skript.
+ *
+ *  Skript is free software: you can redistribute it and/or modify
+ *  it under the terms of the GNU General Public License as published by
+ *  the Free Software Foundation, either version 3 of the License, or
+ *  (at your option) any later version.
+ *
+ *  Skript is distributed in the hope that it will be useful,
+ *  but WITHOUT ANY WARRANTY; without even the implied warranty of
+ *  MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ *  GNU General Public License for more details.
+ *
+ *  You should have received a copy of the GNU General Public License
+ *  along with Skript.  If not, see <http://www.gnu.org/licenses/>.
+ *
+ *
+ * Copyright 2011-2017 Peter Güttinger and contributors
+ */
+package ch.njol.skript.effects;
+
+import java.nio.file.Files;
+import java.nio.file.Path;
+import java.nio.file.Paths;
+
+import org.bukkit.Bukkit;
+import org.bukkit.event.Event;
+import org.bukkit.util.CachedServerIcon;
+import org.eclipse.jdt.annotation.Nullable;
+
+import ch.njol.skript.Skript;
+import ch.njol.skript.doc.Description;
+import ch.njol.skript.doc.Examples;
+import ch.njol.skript.doc.Name;
+import ch.njol.skript.doc.Since;
+import ch.njol.skript.lang.Expression;
+import ch.njol.skript.lang.SkriptParser;
+import ch.njol.skript.util.AsyncEffect;
+import ch.njol.util.Kleenean;
+
+@Name("Load Server Icon")
+@Description({"Loads server icons from the given files. You can get the loaded icon using the <a href='expressions.html#ExprLastLoadedServerIcon'>last loaded server icon</a> expression.",
+		"Please note that the image must be 64x64 and the file path starts from the server folder.",
+		"",
+		"Server icon stuff requires PaperSpigot 1.12.2+."})
+@Examples({"on load:",
+		"	clear {server-icons::*}",
+		"	loop 5 times:",
+		"		load server icon from file \"icons/%loop-number%.png\"",
+		"		add the last loaded server icon to {server-icons::*}",
+		"",
+		"on server list ping:",
+		"	set the icon to a random server icon out of {server-icons::*}"})
+@Since("INSERT VERSION")
+public class EffLoadServerIcon extends AsyncEffect {
+
+	static {
+		Skript.registerEffect(EffLoadServerIcon.class, "(load|cache) [the] server icon (of|from) [image] [file] %string%");
```

Comment:
cache shouldn't really be part of this syntax, because the syntax does not implicitly cache anything. That is up to the user.

Reply:
It caches the file? But yeah i think it is not necessary to reflect this to the user, i will remove "cache".